drgn_tools/util.py

```diff
@@ -123,6 +123,23 @@ def enum_name_get(
     return default
 
 
+def enum_value_get(prog: Program, key: str, offset: int = 0) -> int:
+    """
+    Get the corresponding index of a enum item.
+    If |offset| is not 0, then return the final index as the |key|'s index + |offset|.
+
+    :param prog: drgn program
+    :param key: The target enum item
+    :param offset: The target enum item's offset
+    :returns: The target enum item's index
+    """
+    try:
+        enum_item = prog.constant(key)
+    except LookupError:
+        raise IndexError("UNKNOWN enum %s" % (key))
+    return enum_item.value_() + offset
+
+
```

Comment:
I'm having trouble understanding the purpose of this function. Why use it instead of `int(prog["key"]) + offset`?

It seems like most users don't use the offset anyway.

Reply:
Yes, it is a unified function that checks that the |key| exists in |prog| and also provides the 'offset' functionality. It seems that the offset part is only used in calculating LRU lists' statistics.
I think it makes sense to just calculate the enum value inline. Updated.